lib/orderbook/OrderBook.ts

```diff
@@ -391,14 +392,19 @@ class OrderBook extends EventEmitter {
    * @param quantityToRemove the quantity to remove from the order, if undefined then the full order is removed
    * @returns `true` if the order or portion thereof was removed, otherwise `false`
    */
-  private removePeerOrder = (orderId: string, pairId: string, quantityToRemove?: number) => {
+  private removePeerOrder = (orderId: string, pairId: string, peerPubKey: string, quantityToRemove?: number) => {
     const matchingEngine = this.getMatchingEngine(pairId);
     try {
-      const removeResult = matchingEngine.removePeerOrder(orderId, quantityToRemove);
+      const removeResult = matchingEngine.removePeerOrder(orderId, peerPubKey, quantityToRemove);
       this.emit('peerOrder.invalidation', { orderId, pairId, quantity: removeResult.order.quantity });
       return true;
     } catch (err) {
-      this.logger.error(`attempted to remove non-existing orderId (${orderId})`);
+      let errMsg = `attempted to remove non-existing orderId (${orderId})`;
+      if (err && err.code === errorCodes.PEER_ORDER_NOT_FOUND) {
```

Comment:
at the current situation, you will get this error code only if there are no orders at all for the peer. If there are any, but this specific order is missing, you will get a different error. This is quite confusing and problematic IMO, although at the moment it's affecting just the logging message. 

My suggestion: 
1) `MatchingEngine.removeOrder` should throw the specific order type error (own/peer), or instead `MatchingEngine.removePeerOrder`/`removeOwnOrder` should throw the generic order error. I prefer the former. 
2) I don't think we should use the underlying error message, which doesn't give the full context that we are aware of at this point. I would rather remove the error handling from `OrderBook.removePeerOrder`, and create `OrderBook.handleOrderInvalidation` method (for the `Pool` event listener) to catch the error and to log a better error message.

Reply:
@moshababo thanks for the feedback.

1. Good catch. I think we want to distinguish between own/peer order not found errors. I propose we change the signature of `MatchingEngine.removeOrder` to accept an `order` instead of `orderId` as param. This way we can fetch throw more specific error messages in `MatchingEngine.removePeerOrder` and `MatchingEngine.removeOwnOrder`.

2. Agreed. Will make the changes.

Reply:
For `MatchingEngine.removeOrder` to accept `order instead of `orderId`, I was toying with that a week or two ago and I forget the specifics but I came away thinking it was a lot cleaner to stick to `orderId`. I'm not necessarily opposed, but I remember having this same thought and then thinking better of it.

But I doubt whether we need separate error types for peer or own order not found. Why not just have one, and then let the caller of the function determine how to handle it? The caller will be calling some either `removeOwnOrder` or `removePeerOrder` and therefore will already know if it's a peer order or an own order that wasn't found, even with a single error type for both of them.

Reply:
I think that accepting `order` instead of `orderId` is an overkill to this problem. 

I was thinking of either inspecting the T value somehow (by the injected lists instance, if it's viable), or injecting the error to throw. Other option what @sangaman suggested - to drop the separated errors entirely, and to use only `ORDER_NOW_FOUND`.